worker/manager/src/mirrors_qa_manager/docker.py

```diff
@@ -43,34 +45,36 @@ def wrapped(*args: Any, **kwargs: Any):
 
         return wrapped
 
+    if func:
```

Comment:
This is a decorator? Why should we care about it being called without a func? It's purpose is to decorate a func

Reply:
Um, that's what enables the bare `@retry` to work even when it's written as `@retry()`
In the initial PR that didn't have this check, I had to do `@retry()` on every function that used it and that had only `retries` and `interval` as the parameter. There was no `func` as the first argument to `def retry`.

The `retry` function isn't a decorator in its own rather it returns a decorator. And in the first PR, I had to invoke it explicitly  so it could return the actual decorator while still maintaining closure over the default parameters. 

In the changes you requested for, you said to change it to be `@retry` and that meant not invoking the decorator in which case, I had to do this check as the function being decorated wouldn't be passed. I could switch back to the older style as I think it's more explicit.

Reply:
Ah indeed you return a decorator. I don't see what value it brings here. Am I missing something?
We can figure it out later ; no reason to slowdown the PR f this

Reply:
Okay